django/chat/responses.py

```diff
@@ -55,25 +51,32 @@ def otto_response(request, message_id=None):
 
 def chat_response(chat, response_message, eval=False):
 
-    user_message = response_message.parent
+    def is_text_to_summarize(message):
+        return message.mode == "summarize" and not message.is_bot
 
     system_prompt = chat.options.chat_system_prompt
-    chat_history = [SystemMessage(content=system_prompt)]
+    chat_history = [ChatMessage(role=MessageRole.SYSTEM, content=system_prompt)]
     chat_history += [
-        (
-            AIMessage(content=message.text)
-            if message.is_bot
-            else HumanMessage(content=message.text)
+        ChatMessage(
+            role=MessageRole.ASSISTANT if message.is_bot else MessageRole.USER,
+            content=(
+                message.text
+                if not is_text_to_summarize(message)
+                else "<text to summarize...>"
```

Comment:
This is to prevent long texts (which user input for summarization purposes) to be passed into Chat mode later, if the mode is switched from Summarize to Chat.